app/src/main/java/com/hedvig/app/feature/keygear/KeyGearValuationActivity.kt

```diff
@@ -28,16 +40,42 @@ class KeyGearValuationActivity : BaseActivity(R.layout.activity_key_gear_valuati
         }
 
         continueButton.setHapticClickListener {
-            //TODO
-            model.submit()
-            onBackPressed()
+            val price = priceInput.getText()
+
+            safeLet(date, id) { date, id ->
+                val monetaryValue =
+                    MonetaryAmountV2Input.builder().amount(price).currency("SEK").build()
+                model.updatePurchaseDateAndPrice(id, date, monetaryValue)
+                onBackPressed()
```

Comment:
What action we take here should not be `onBackPressed()` in the future, hence we should add a TODO here explaining that.

I think it's probably also appropriate to add the transitionary state here with the loading button 🙂